soccerdata/sofascore.py

```diff
@@ -0,0 +1,265 @@
+              be called after failed requests, allowing rotating proxies.
+    no_cache : bool
+        If True, will not use cached data.
+    no_store : bool
+        If True, will not store downloaded data.
+    data_dir : Path
+        Path to directory where data will be cached.
+    """
+
+    def __init__(
+        self,
+        leagues: Optional[Union[str, List[str]]] = None,
+        seasons: Optional[Union[str, int, Iterable[Union[str, int]]]] = None,
+        proxy: Optional[
+            Union[str, Dict[str, str], List[Dict[str, str]], Callable[[], Dict[str, str]]]
+        ] = None,
+        no_cache: bool = NOCACHE,
+        no_store: bool = NOSTORE,
+        data_dir: Path = SOFASCORE_DATADIR,
+    ):
+        """Initialize the Sofascore reader."""
+        super().__init__(
+            leagues=leagues,
+            proxy=proxy,
+            no_cache=no_cache,
+            no_store=no_store,
+            data_dir=data_dir,
+        )
+        self.seasons = seasons  # type: ignore
+        if not self.no_store:
+            (self.data_dir / "leagues").mkdir(parents=True, exist_ok=True)
+            (self.data_dir / "seasons").mkdir(parents=True, exist_ok=True)
+            (self.data_dir / "matches").mkdir(parents=True, exist_ok=True)
+
+    def read_leagues(self) -> pd.DataFrame:
+        """Retrieve the selected leagues from the datasource.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        url = SOFASCORE_API + 'config/top-unique-tournaments/EN/football'
+        filepath = self.data_dir / 'leagues.json'
+        reader = self.get(url, filepath)
+        data = json.load(reader)
+        leagues = []
+        for k in data['uniqueTournaments']:
+            if k['name'] in BIG_FIVE_DICT:
+                leagues.append(
+                    {
+                        'region': BIG_FIVE_DICT[k['name']].split("-")[0],
+                        'league_id': k['id'],
+                        'league': BIG_FIVE_DICT[k['name']],
+                    }
+                )
+        df = pd.DataFrame(leagues).set_index('league')
+        return df[df.index.isin(self.leagues)]
+
+    def read_seasons(self) -> pd.DataFrame:
+        """Retrieve the selected seasons for the selected leagues.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        filemask = 'leagues/{}.json'
+        seasons = []
+        df_leagues = self.read_leagues()
+        for lkey, league in df_leagues.iterrows():
+            print
+            url = SOFASCORE_API + 'unique-tournament/{}/seasons'
+            filepath = self.data_dir / filemask.format(lkey)
+            reader = self.get(url.format(league.league_id), filepath)
+            data = json.load(reader)['seasons']
+            for season in data:
+                seasons.append(
+                    {
+                        'league': lkey,
+                        'season': season['year'].replace("/", ""),
```

Comment:
Use the `season_code` helper to convert seasons. This ensures that season codes are consistent everywhere.

Reply:
Included now